src/item/SmithingTemplate.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+class SmithingTemplate extends Item{
```

Comment:
Not needed if its empty

Reply:
iirc it was made to determine the smithing template from other inputs in transaction

Reply:
Use TypeId ?

Reply:
> Use TypeId ?

https://github.com/pmmp/PocketMine-MP/pull/6202#discussion_r1432552692

Reply:
I don't think creating a new empty class just for that is viable. Another way can be done, i'm sure.

Reply:
if you propose one, I'd be happy aswell

Reply:
I think you can use ```match()``` or ```ArmorTrimPattern::fromItem()```. What do you think about that @ShockedPlot7560?

Reply:
fromItem could be a solution, however using gameplay API for some network validation, i dont know